src/simsopt/geo/magneticfield.py

```diff
@@ -1,139 +1,133 @@
+        """Export the field evaluated on a regular grid for visualisation with e.g. Paraview."""
+        from pyevtk.hl import gridToVTK
+        rs = np.linspace(rmin, rmax, nr, endpoint=True)
+        phis = np.linspace(0, 2*np.pi, nphi, endpoint=True)
+        zs = np.linspace(zmin, zmax, nz, endpoint=True)
+
+        R, Phi, Z = np.meshgrid(rs, phis, zs)
+        X = R * np.cos(Phi)
+        Y = R * np.sin(Phi)
+        Z = Z
+
+        RPhiZ = np.zeros((R.size, 3))
+        RPhiZ[:, 0] = R.flatten()
+        RPhiZ[:, 1] = Phi.flatten()
+        RPhiZ[:, 2] = Z.flatten()
+
+        self.set_points_cyl(RPhiZ)
+        vals = self.B()
+        contig = np.ascontiguousarray
+        gridToVTK(filename, X, Y, Z, pointData={"B": (contig(vals[:, 0]), contig(vals[:, 1]), contig(vals[:, 2]))})
+
 
 class MagneticFieldMultiply(MagneticField):
-    '''
-    Class used to multiply a magnetic field by a scalar.
-    It takes as input a MagneticField class and a scalar and multiplies B, A and their derivatives by that value
-    '''
+    """
+    Class used to multiply a magnetic field by a scalar.  It takes as input a
+    MagneticField class and a scalar and multiplies B, A and their derivatives
+    by that value.
+    """
 
     def __init__(self, scalar, Bfield):
+        MagneticField.__init__(self)
         self.scalar = scalar
         self.Bfield = Bfield
 
-    def set_points(self, points):
-        self.points = np.array(points)
-        self.clear_cached_properties()
-        self.Bfield.set_points(points)
-        return self
-
-    def compute(self, points, compute_derivatives=0):
-        self.Bfield.compute(points, compute_derivatives)
-        self._B = self.scalar*self.Bfield._B
-        if compute_derivatives >= 1:
-            self._dB_by_dX = self.scalar*self.Bfield._dB_by_dX
-        if compute_derivatives >= 2:
-            self._d2B_by_dXdX = self.scalar*self.Bfield._d2B_by_dXdX
-
-    def compute_A(self, points, compute_derivatives=0):
-        self.Bfield.compute_A(points, compute_derivatives)
-        self._A = self.scalar*self.Bfield._A
-        if compute_derivatives >= 1:
-            self._dA_by_dX = self.scalar*self.Bfield._dA_by_dX
-        if compute_derivatives >= 2:
-            self._d2A_by_dXdX = self.scalar*self.Bfield._d2A_by_dXdX
+    def _set_points_cb(self):
+        self.Bfield.set_points_cart(self.get_points_cart_ref())
+
+    def _B_impl(self, B):
+        B[:] = self.scalar*self.Bfield.B()
+
+    def _dB_by_dX_impl(self, dB):
+        dB[:] = self.scalar*self.Bfield.dB_by_dX()
+
+    def _d2B_by_dXdX_impl(self, ddB):
+        ddB[:] = self.scalar*self.Bfield.d2B_by_dXdX()
+
+    def _A_impl(self, A):
+        A[:] = self.scalar*self.Bfield.A()
+
+    def _dA_by_dX_impl(self, dA):
+        dA[:] = self.scalar*self.Bfield.dA_by_dX()
+
+    def _d2A_by_dXdX_impl(self, ddA):
+        ddA[:] = self.scalar*self.Bfield.d2A_by_dXdX()
 
 
 class MagneticFieldSum(MagneticField):
```

Comment:
Same question for this class as well?

Reply:
Discussed on slack. For future reference: the point is to support lazy evaluation of expressions such as `B = B0 + alpha * B1`